src/double_linked_list.py

```diff
@@ -0,0 +1,104 @@
+"""Creats a double linked list."""
+
+
+class DoubleLinkedList(object):
+    """A singly-linked list."""
+
+    def __init__(self, data=None):
+        """Create an instance of type LinkedList. Allow data to be passed in."""
+        self.head = None
+        self.tail = None
+        if data is not None:
+            try:
+                for item in data:
+                    if item is data[0]:
+                        self.head = Node(item, next=None, prev=None)
+                        self.tail = self.head
+                    else:
+                        node = Node(item, self.head, prev=None)
+                        self.head.prev = node
+                        self.head = node
+            except TypeError:
+                node = Node(data, next=None, prev=None)
+                self.head = node
+                self.tail = self.head
+
+
+    def push(self, val):
+        """Push a val to the end of the list."""
+        node = Node(val, self.head, None)
+        if self.head is not None:
+            self.head.prev = node
+        self.head = node
+
+
+    def append(self, val):
+        """Append a val to the  start of the list."""
+        node = Node(val, None, self.tail)
+        if self.tail is not None:
+            self.tail.next = node
+        self.tail = node
+
+
+    def pop(self):
+        try:
+            val = self.head.data
+            if self.head.next == None:
+                self.tail = None
+            self.head = self.head.next
+            self.head.prev = None
+            return val
+        except AttributeError:
+            raise AttributeError("Cannot pop from an empty list.")
+    
+
+    def shift(self):
+        try:
+            val = self.tail.data
+            if self.tail.prev == None:
+                self.head = None
+            self.tail = self.tail.prev
+            self.tail.prev = None
```

Comment:
This is going to throw you a bug. You need to make it so that `tail.next = None`